src/Dotnet.Script/Program.cs

```diff
@@ -83,13 +83,9 @@ private static int Wain(string[] args)
                 c.HelpOption(helpOptionTemplate);
                 c.OnExecute(async () =>
                 {
-                    if (string.IsNullOrWhiteSpace(code.Value))
-                    {
-                        c.ShowHelp();
-                        return 0;
-                    }
+                    var source = string.IsNullOrWhiteSpace(code.Value) ? await Console.In.ReadToEndAsync() : code.Value;
```

Comment:
this is a nice idea, but what if the piped input is not there? it just hangs?

Reply:
Yes, just like `cat` and many other tools that default to receiving content over standard input. That said, I can imagine it being confusing for many. Having the ability to interactively type the input isn't going to be the common case and the experience varies sufficiently (e.g. line editing & EOF) between terminals and platforms that it will only add to the confusion. I have therefore made an update with ee28276fb32ee6d4bfddea24b0442e9166f28360 where [`Console.IsInputRedirected`](https://docs.microsoft.com/en-us/dotnet/api/system.console.isinputredirected) is used to detect if standard input has been redirected and display help otherwise. So now, we are back to the current behaviour where `dotnet script eval` alone will display help, but if there's code provided as an argument or supplied over standard input, then it's executed.